guide/creating-your-bot/command-deployment.md

```diff
@@ -0,0 +1,126 @@
+
+If you already have slash commands set up and deployed for your application and want to learn how to respond to them, refer to the following section on [replying to slash commands](/slash-commands/replying.md).
+:::
+
+In this section, we'll cover how to register your commands to Discord using discord.js!
+
+## Command registration
+
+Slash commands can be registered in two ways; in one specific guild, or for every guild the bot is in. We're going to look at single-guild registration first, as this is a good way to develop and test your commands before a global deployment.
+
+Your application will need the `applications.commands` scope authorized in a guild for either its global or guild slash commands to appear, and to register them in a specific guild without error.
+
+Slash commands only need to be registered once, and updated when the definition (description, options etc) is changed. As there is a daily limit on command creations, it's not necessary nor desirable to connect a whole client to the gateway or do this on every `ready` event. As such, a standalone script using the lighter REST manager is preferred. 
+
+This script is intended to be run separately, only when you need to make changes to your slash command **definitions** - you're free to modify parts such as the execute function as much as you like without redeployment. 
+
+### Guild commands
+
+Create a `deploy-commands.js` file in your project directory. This file will be used to register and update the slash commands for your bot application.
+
+Add two more properties to your `config.json` file, which we'll need in the deployment script:
+
+- `clientId`: Your application's client id
+- `guildId`: Your development server's id
+
+```json
+{
+	"token": "your-token-goes-here",
+	"clientId": "123456789012345678",
+	"guildId": "876543210987654321",
+}
+```
+
+With these defined, you can use the deployment script below:
+
+<!-- eslint-skip -->
+
+```js
+const { REST, Routes } = require('discord.js');
+const { token } = require('./config.json');
+const fs = require('node:fs');
+
+const commands = [];
+// Grab all the command files from the commands directory you created earlier
+const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));
+
+// Place your client and guild ids here
+const clientId = '123456789012345678';
+const guildId = '876543210987654321';
+
+// Grab the SlashCommandBuilder#toJSON() output of each command's data for deployment
+for (const file of commandFiles) {
+	const command = require(`./commands/${file}`);
+	commands.push(command.data.toJSON());
+}
+
+// Construct and prepare an instance of the REST module
+const rest = new REST({ version: '10' }).setToken(token);
+
+// and deploy your commands!
+(async () => {
```

Comment:
toplevel await should be available in any node version compatible with v14
this section is also not consistent with code samples (log lines, etc.)

Reply:
Isn't this only true if it's an ESM module?